src/main/java/com/avioconsulting/mule/logger/internal/CustomLoggerOperation.java

```diff
@@ -2,31 +2,132 @@
+import org.mule.runtime.core.api.util.IOUtils;
 import org.mule.runtime.extension.api.annotation.param.Config;
 import org.mule.runtime.extension.api.annotation.param.MediaType;
 import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+import org.mule.runtime.extension.api.client.DefaultOperationParameters;
+import org.mule.runtime.extension.api.client.DefaultOperationParametersBuilder;
+import org.mule.runtime.extension.api.client.ExtensionsClient;
+import org.mule.runtime.extension.api.runtime.operation.Result;
 import org.mule.runtime.extension.api.runtime.parameter.CorrelationInfo;
+import org.mule.runtime.extension.api.runtime.parameter.ParameterResolver;
+import org.mule.runtime.extension.api.runtime.streaming.StreamingHelper;
 
 /**
  * Author: Chakri Bonthala
  */
 public class CustomLoggerOperation {
 
+  @Inject
+  ExtensionsClient extensionsClient;
+
   @MediaType(value = ANY, strict = false)
   public void log(@ParameterGroup(name = "Log") LogProperties logProperties,
       @ParameterGroup(name = "Message Attributes") MessageAttributes messageAttributes,
       @ParameterGroup(name = "Exception Details") ExceptionProperties exceptionProperties,
       @ParameterGroup(name = "Additional Options") AdditionalProperties additionalProperties,
       @Config CustomLoggerConfiguration loggerConfig,
       ComponentLocation location,
-      CorrelationInfo correlationInfo) {
+      CorrelationInfo correlationInfo,
+      StreamingHelper streamingHelper) {
+
+    transformPayload(logProperties, loggerConfig, streamingHelper);
     loggerConfig.getLogger().log(logProperties, messageAttributes, exceptionProperties, additionalProperties,
         loggerConfig, location, correlationInfo.getCorrelationId());
   }
+
+  byte[] convertToByteArray(Object payload, StreamingHelper streamingHelper) {
+    Object resolved = streamingHelper.resolveCursorProvider(payload);
+    if (resolved instanceof CursorProvider) {
+      CursorStream cursorStream = ((CursorProvider<CursorStream>) resolved).openCursor();
+      return IOUtils.toByteArray(cursorStream);
+    } else {
+      return null;
+    }
+  }
+
+  void transformPayload(LogProperties logProperties, CustomLoggerConfiguration loggerConfig,
+      StreamingHelper streamingHelper) {
+    ParameterResolver<String> payload = logProperties.getPayload();
+    if (payload == null)
+      return;
+    Result<InputStream, Void> executeCompress = null;
+    executeCompress = setCompressedPayloadIfNeeded(logProperties, streamingHelper, loggerConfig);
+    setEncryptedPayloadIfNeeded(logProperties, loggerConfig, streamingHelper, executeCompress);
+  }
+
+  Result<InputStream, Void> setCompressedPayloadIfNeeded(LogProperties logProperties,
+      StreamingHelper streamingHelper,
+      CustomLoggerConfiguration loggerConfig) {
+    /*
+     * If compressor is not null, compress payload with provided compressor strategy
+     * as of 12/02/22, GZIP is the only compressor supported
+     */
```

Comment:
This comment shouldn't be here, support is driven by the values in the `Compressor` Enum class.